src/main/java/lotto/domain/LottoNumber.java

```diff
@@ -1,34 +1,46 @@
 package lotto.domain;
 
 import java.util.Objects;
-import java.util.regex.Pattern;
 import lotto.utils.CustomException;
+import lotto.utils.StringChecker;
 
 public class LottoNumber {
     public static final int MINIMUM_NUMBER = 1;
     public static final int MAXIMUM_NUMBER = 45;
-    private static final String IS_NUMBER = "\\d+";
+    private static final LottoNumber[] cache = new LottoNumber[MAXIMUM_NUMBER + 1];
 
     private final int number;
 
-
-    public LottoNumber(String numberValue) {
-        this(convertToLottoNumber(numberValue));
+    static {
+        for (int i = 1; i <= MAXIMUM_NUMBER; i++) {
+            cache[i] = new LottoNumber(i);
+        }
```

Comment:
```suggestion
    static {
        for (int i = 0; i < MAXIMUM_NUMBER; i++) {
            cache[i] = new LottoNumber(i + MINIMUM_NUMBER);
        }
```

큰 문제는 아니지만 개인적으론 0번 인덱스부터 사용하는게 좋을 것 같다는 생각이 드는데요!
1부터 생성하였을 때 어떠한 이점을 얻을 수 있을까요?
지금은 시작이 1이지만 너 큰 숫자가 온다면 불필요한 메모리 낭비가 될 것 같습니다 ㅎㅎ

사실 해당 코멘트는 저도 애매한 부분이 있어서 의견을 나눠보고 싶네요 :)

Reply:
1부터 생성하였을 땐, 숫자1은 1번 인덱스가 되어서 인덱스와 숫자를 일치시켜서 덜 헷갈리게 하려고 했었어요. (항상 인덱스 접근할 때, 실수하는 경우가 많아서요ㅠㅠ) 하지만 재연링이 말해준대로 지금이야 1부터 시작하니 1개의 배열 크기만 낭비되지만, 시작 숫자가 커지면 낭비가 많이 될 듯 하네요! 😀

이 부분을 고치면서 든 생각은 index와 숫자가 일치하지 않을 경우에 내부 값도 의도된 대로 지정되었는지 확실하게 테스트를 진행해줘야 할 것 같아요! new LottoNumber(1)로 해서 객체비교가 아닌 getNumber로 내부 숫자 값 자체를 비교하는 테스트코드를 예전에 만들었었는데, 해당 테스트 코드가 실패가 떠서 수정할 수 있었어요. 해당 테스트 코드가 없었으면 실수를 확인하지 못하고 넘어갔을 것 같아요! 😥 

하지만 또 드는 생각은 LottoNumber가 VO 객체인데 getter로 값 자체를 꺼내서 볼 수 있게 하는게 좋은걸까? 하는 의문도 들기 시작하네요. 여기에 대한 재연링의 의견이 궁금합니다. 👀

Reply:
수업을 듣고 cache을 Map 자료구조로 변경해서 인덱스 혼란 요소를 줄여주었어요! getter를 없애려고 해봤는데, View에서 내부 값을 보여줘야해서 view랑 연결되면 getter를 지양하기 어렵지 않을까 생각했어요. 😥